### PR TITLE
Implement some rules from `flake8-logging-format`

### PR DESCRIPTION
**Pull request by edgarrmondragon**
https://github.com/charliermarsh/ruff/issues/1850

### CODE REVIEW

File: src/rules/flake8_logging_format/rules.rs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                // G001 - G004
+                if let Some(format_arg) = call_args.get_argument("msg", Some(0)) {
+                    check_msg(checker, format_arg);
+                }
+
+                // G010
+                if checker.settings.rules.enabled(&Rule::LoggingWarn)
+                    && matches!(logging_level, LoggingLevel::Warn)
+                {
+                    let mut diagnostic = Diagnostic::new(LoggingWarn, level_call_range);
```
Comment (charliermarsh) [inline src/rules/flake8_logging_format/rules.rs:185]: One concern here is that `PGH002` does the same thing:

```
resources/test/fixtures/flake8_logging_format/G010.py:3:1: PGH002 `warn` is deprecated in favor of `warning`
```

We could consider deprecating that rule, though, in favor of this more comprehensive plugin. We'd basically just redirect `PGH002` to `G010`.

File: src/rules/flake8_logging_format/rules.rs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                }
+            }
+        }
+        _ => {}
+    }
+}
+
+/// Check logging calls for violations
+pub fn logging_call(checker: &mut Checker, func: &Expr, args: &[Expr], keywords: &[Keyword]) {
+    if let ExprKind::Attribute { value, attr, .. } = &func.node {
```
Comment (charliermarsh) [inline src/rules/flake8_logging_format/rules.rs:157]: In `src/rules/tryceratops/rules/error_instead_of_exception.rs`, I used some heuristics to detect whether the `value` is that of a logger (by matching on names, like `logging`, `logger`, `self.logger`, `foo_logger`, etc.). I'd like to be consistent between that rule and here. Any opinion on what's preferable?

File: src/rules/flake8_logging_format/rules.rs [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                                Range::from_located(key),
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+        ExprKind::Call { func, keywords, .. } => {
+            if let ExprKind::Name { id, .. } = &func.node {
+                if id == "dict" {
```
Comment (charliermarsh) [inline src/rules/flake8_logging_format/rules.rs:137]: Let's use `checker.resolve_call_path(expr).map_or(false, |call_path| call_path.as_slice() == ["", "dict"])` here instead. That way, it'll be robust to overrides on `dict` (not common, but good to be consistent).

Comment (edgarrmondragon) [reply src/rules/flake8_logging_format/rules.rs]: Makes sense, thanks for the suggestion. Done: ad0e93bf5a2a075bedef3a8377a1378ff65a4ad2

File: src/rules/flake8_logging_format/rules.rs (hunk 2 above)
Comment (edgarrmondragon) [reply src/rules/flake8_logging_format/rules.rs:158]: I followed the plugin here to just whitelist `parser` and `warnings` but it probably makes sense to make this more robust with some `is_logger_candidate` that incorporates your heuristic additionally. Happy to try out the refactor.

Comment (charliermarsh) [reply src/rules/flake8_logging_format/rules.rs:158]: That would be great, thanks!

File: src/rules/flake8_logging_format/rules.rs (hunk 1 above)
Comment (edgarrmondragon) [reply src/rules/flake8_logging_format/rules.rs:186]: I agree that it makes sense to prefer the logging plugin for this rule. Is it just a matter of adding it in `src/rule_redirects.rs`?

File: src/rules/flake8_logging_format/rules.rs (hunk 2 above)
Comment (edgarrmondragon) [reply src/rules/flake8_logging_format/rules.rs:156]: [29b19c6](https://github.com/charliermarsh/ruff/pull/2150/commits/29b19c6c554052dd63cf23c7e4ba0bf69894b729)

File: src/rules/flake8_logging_format/rules.rs (hunk 1 above)
Comment (charliermarsh) [reply src/rules/flake8_logging_format/rules.rs]: Yeah, we'd need to delete all references to the rule (fixture files, etc.), then add the redirect to that map.

**Pull request merged**